qendpoint-store/src/main/java/com/the_qa_company/qendpoint/utils/VariableToIdSubstitution.java

```diff
@@ -63,7 +63,9 @@ public void meet(Var var) {
 					}
 				}
 				if (id != -1) {
-					var.setValue(converter.idToHDTValue(id, position));
+					Var var1 = new Var(var.getName(), converter.idToHDTValue(id, position), var.isAnonymous(),
+							var.isConstant());
+					var.replaceWith(var1);
```

Comment:
In order to reduce bugs in RDF4J we now require that Var objects are not reused.